skglm/solvers/fista.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from numba import njit
+from skglm.solvers.base import BaseSolver
+
+
+@njit
+def _prox_vec(w, z, penalty, lipschitz):
+    # XXX: TO DISCUSS: should add a vectorized prox update
+    n_features = w.shape[0]
+    for j in range(n_features):
+        w[j] = penalty.prox_1d(z[j], 1 / lipschitz, j)
+    return w
+
+
+class FISTA(BaseSolver):
+    r"""ISTA solver with Nesterov acceleration (FISTA)."""
+
+    def __init__(self, max_iter=100, tol=1e-4, fit_intercept=False, warm_start=False,
+                 opt_freq=100, verbose=0):
```

Comment:
It doesn't seem other solvers have `opt_freq` so there's no comparison, but why 100 here ?

Reply:
I used `opt_freq` to avoid computing the optimality conditions at every FISTA iteration. I know we don't implement this in other solvers but it feels like an outer iteration of WS + CD is not the same as FISTA. For 100, purely "au doigt mouillé".

Reply:
maybe 10 instead of 100 ? cost is the same as 1 single iteration, 10 % seems reasonable